## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 454315`
Commit: [`90579cacab21c8ca8af81017afcdd31a6caa27eb`](https://github.com/NixOS/nixpkgs/commit/90579cacab21c8ca8af81017afcdd31a6caa27eb) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/90579cacab21c8ca8af81017afcdd31a6caa27eb..pull/454315/head))
Merge: [`98ab1b4085dd40ec8d8f39c0d12754fabeec13de`](https://github.com/NixOS/nixpkgs/commit/98ab1b4085dd40ec8d8f39c0d12754fabeec13de)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/18696639252


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>liblaxjson</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>liblaxjson</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>liblaxjson</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>liblaxjson</summary>
<pre>build flags: -j3 SHELL=/nix/store/r48xc0431lrq2xaxnwq6f0iwd88373cy-bash-5.3p3/bin/bash
[ 12%] Building C object CMakeFiles/laxjson.dir/src/laxjson.c.o
[ 25%] Building C object CMakeFiles/laxjson_static.dir/src/laxjson.c.o
[ 37%] Linking C shared library liblaxjson.dylib
[ 50%] Linking C static library liblaxjson.a
[ 50%] Built target laxjson_static
[ 50%] Built target laxjson
[ 62%] Building C object CMakeFiles/token_list.dir/example/token_list.c.o
[ 75%] Building C object CMakeFiles/primitives_test.dir/test/primitives.c.o
/nix/build/nix-32461-4294621323/source/example/token_list.c:53:9: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
   53 | int main() {
      |         ^
      |          void
1 error generated.
make[2]: *** [CMakeFiles/token_list.dir/build.make:79: CMakeFiles/token_list.dir/example/token_list.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:160: CMakeFiles/token_list.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 87%] Linking C executable primitives_test
[ 87%] Built target primitives_test
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>liblaxjson</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>liblaxjson</summary>
<pre>build flags: -j3 SHELL=/nix/store/k8a78r7j84kh8vagydpb8zwaqnjch8fd-bash-5.3p3/bin/bash
[ 25%] Building C object CMakeFiles/laxjson_static.dir/src/laxjson.c.o
[ 25%] Building C object CMakeFiles/laxjson.dir/src/laxjson.c.o
[ 50%] Linking C shared library liblaxjson.dylib
[ 50%] Linking C static library liblaxjson.a
[ 50%] Built target laxjson_static
[ 50%] Built target laxjson
[ 75%] Building C object CMakeFiles/token_list.dir/example/token_list.c.o
[ 75%] Building C object CMakeFiles/primitives_test.dir/test/primitives.c.o
/nix/build/nix-5643-2179729919/source/example/token_list.c:53:9: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
   53 | int main() {
      |         ^
      |          void
1 error generated.
make[2]: *** [CMakeFiles/token_list.dir/build.make:79: CMakeFiles/token_list.dir/example/token_list.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:160: CMakeFiles/token_list.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 87%] Linking C executable primitives_test
[ 87%] Built target primitives_test
make: *** [Makefile:146: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/li/liblaxjson/package.nix
+++ b/pkgs/by-name/li/liblaxjson/package.nix
@@ -18,11 +18,17 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 2.8)" "cmake_minimum_required(VERSION 3.10)"
+  '';
+
   meta = with lib; {
     description = "Library for parsing JSON config files";
     homepage = "https://github.com/andrewrk/liblaxjson";
     license = licenses.mit;
     platforms = platforms.unix;
+    broken = stdenv.hostPlatform.isDarwin;
     maintainers = [ maintainers.andrewrk ];
   };
 }

